pkg/wharfapi/build.go

```diff
@@ -29,3 +34,67 @@ type BuildParam struct {
 	Name    string `json:"name"`
 	Value   string `json:"value"`
 }
+
+// GetBuildList gets all builds by invoking the HTTP request:
+//  GET /api/build
+func (c Client) GetBuildList() ([]response.Build, error) {
+	path := "/api/build"
+	list := []response.Build{}
+	err := c.PutDecoded(&list, "BUILD", path, nil, nil)
+	if err != nil {
+		return []response.Build{}, err
+	}
+	return list, nil
+}
+
+// GetBuild gets a build by invoking the HTTP request:
+//  GET /api/build/{buildId}
+func (c Client) GetBuild(buildID uint) (response.Build, error) {
+	path := fmt.Sprintf("/api/build/%d", buildID)
+	build := response.Build{}
+	err := c.GetDecoded(&build, "BUILD", path, nil)
+	return build, err
+}
+
+// UpdateBuildStatus updates a build by invoking the HTTP request:
+// 	PUT /api/build/{buildId}/status
+func (c Client) UpdateBuildStatus(buildID uint, status request.LogOrStatusUpdate) (response.Build, error) {
+	updatedBuild := response.Build{}
+	body, err := json.Marshal(&status)
+	if err != nil {
+		return updatedBuild, err
+	}
+
+	path := fmt.Sprintf("/api/build/%d/status", buildID)
+	err = c.PutDecoded(&updatedBuild, "BUILD", path, nil, body)
+	return updatedBuild, err
+}
+
+// CreateBuildLog adds a new log to a build by invoking the HTTP request:
+// 	POST /api/build/{buildId}/log
+func (c Client) CreateBuildLog(buildID uint, buildLog request.LogOrStatusUpdate) error {
+	body, err := json.Marshal(buildLog)
+	if err != nil {
+		return err
+	}
+
+	path := fmt.Sprintf("/api/build/%d/log", buildID)
+	_, err = c.Post("BUILD", path, nil, body)
+	return err
```

Comment:
You're marshalling in a lot of functions. Suggest de-duplicating by adding methods such as:

```go
func (c Client) Post(from, path string, q url.Values, body []byte) ([]byte, error) {}
func (c Client) PostJSON(from, path string, q url.Values, body interface{}) ([]byte, error) {}
func (c Client) PostJSONDecoded(from, path string, q url.Values, body interface{}, response interface{}) ([]byte, error) {}
```